rotkehlchen/chain/ethereum/modules/aave/v2/decoder.py

```diff
@@ -0,0 +1,253 @@
+                    event.notes = notes
+                    event.counterparty = CPT_AAVE_V2
+                else:
+                    event.event_subtype = HistoryEventSubType.RECEIVE_WRAPPED
+                    resolved_asset = event.asset.resolve_to_asset_with_symbol()
+                    event.notes = f'Receive {event.balance.amount} {resolved_asset.symbol} from AAVE v2'  # noqa: E501
+                    event.counterparty = CPT_AAVE_V2
+                # Set counterparty for both events
+                event.counterparty = CPT_AAVE_V2
+
+    def _decode_withdrawal(
+            self,
+            token: 'EvmToken',
+            tx_log: EvmTxReceiptLog,
+            decoded_events: list[HistoryBaseEntry],
+    ) -> None:
+        """Decode aave v2 withdrawal event"""
+        user = hex_or_bytes_to_address(tx_log.topics[2])
+        to = hex_or_bytes_to_address(tx_log.topics[3])
+        amount = asset_normalized_value(
+            amount=hex_or_bytes_to_int(tx_log.data),
+            asset=token,
+        )
+        notes = f'Withdraw {amount} {token.symbol} from AAVE v2 user {user}'
+        if to != user:
+            notes += f' to {to}'
+        for event in decoded_events:
+            if (
+                event.balance.amount == amount and
+                event.event_subtype == HistoryEventSubType.NONE
+            ):
+                if (
+                    event.location_label == to and
+                    event.event_type == HistoryEventType.RECEIVE
+                ):
+                    event.event_type = HistoryEventType.WITHDRAWAL
+                    event.event_subtype = HistoryEventSubType.REMOVE_ASSET
+                    event.notes = notes
+                else:
+                    event.event_subtype = HistoryEventSubType.RETURN_WRAPPED
+                    resolved_asset = event.asset.resolve_to_asset_with_symbol()
+                    event.notes = f'Return {amount} {resolved_asset.symbol} to AAVE v2'  # noqa: E501
+                # Set counterparty for both events
+                event.counterparty = CPT_AAVE_V2
+
+    def _decode_borrow(
+            self,
+            token: 'EvmToken',
+            tx_log: EvmTxReceiptLog,
+            decoded_events: list[HistoryBaseEntry],
+    ) -> None:
+        """Decode aave v2 borrow event"""
+        on_behalf_of = hex_or_bytes_to_address(tx_log.topics[2])
+        user = hex_or_bytes_to_address(tx_log.data[:32])
+        amount = asset_normalized_value(
+            amount=hex_or_bytes_to_int(tx_log.data[32:64]),
+            asset=token,
+        )
+        rate_mode = hex_or_bytes_to_int(tx_log.data[64:96])
+        rate = token_normalized_value_decimals(
+            token_amount=hex_or_bytes_to_int(tx_log.data[96:128]),
+            token_decimals=25,
+        )
+        notes = f'Borrow {amount} {token.symbol} from AAVE v2 user {user} with {"stable" if rate_mode == 1 else "variable"} APY {rate}'  # noqa: E501
```

Comment:
the user does not need to be shown here as it's the location label. Since we have the behalf of here perhaps figure out which is the location label and only show the other. (applies to the if below too)